azure/defaults.go

```diff
@@ -380,11 +379,7 @@ func ARMClientOptions(azureEnvironment string, extraPolicies ...policy.Policy) (
 	opts.PerCallPolicies = append(opts.PerCallPolicies, extraPolicies...)
 	opts.Retry.MaxRetries = -1 // Less than zero means one try and no retries.
 
-	otelTP, err := ot.OTLPTracerProvider(context.TODO())
-	if err != nil {
-		return nil, err
-	}
-	opts.TracingProvider = azotel.NewTracingProvider(otelTP, nil)
+	opts.TracingProvider = azotel.NewTracingProvider(otel.GetTracerProvider(), nil)
```

Comment:
Do these reflect changing things that were first introduced in 1.18? As far as I can tell these changes were introduced in the v1.17.0 release:

- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/5144/files#diff-993cbd654628d8e1f18959c6e6ce555d7f3a96408639ddec70a62004666f68a5

cc @nojnhuh

Reply:
That PR only exists for 1.18 not backported AFAICT: https://github.com/kubernetes-sigs/cluster-api-provider-azure/commit/abd88192b102fef877d702376d1ff3baa807895f

> I don't really know if this is all totally "correct" or "the right way to do things"

Turns out it is neither. This looks much better! I can still see the traces.

/lgtm